src/module.ts

```diff
@@ -29,13 +31,24 @@ export default defineNuxtModule<ModuleOptions>({
   },
   async setup(options, nuxt) {
     const resolver = createResolver(import.meta.url)
+    // opt-out early
+    if (!options.enabled) {
+      return
+    }
 
     // add options to runtime config
-    nuxt.options.runtimeConfig.public.nuxtRateLimit = defu(
-      nuxt.options.runtimeConfig.public.nuxtRateLimit,
-      options
+    nuxt.options.runtimeConfig.nuxtRateLimit = defu(
+      nuxt.options.runtimeConfig.nuxtRateLimit,
+      { headers: options.headers }
     )
 
+    // merge with route rules, so we get free route matching with baseURL support
```

Comment:
Using route rules instead of implementing them yourself gives you a bunch of stuff for free:
- handling base URL from nitro
- lookup route rules based on the request path `getRouteRules`
- auto-merging and server access
- support via macros (defineRouteRules on page files - coming soon).